neo/rawio/spikeglxrawio.py

```diff
@@ -0,0 +1,315 @@
+                global_chan += 1
+
+        sig_channels = np.array(sig_channels, dtype=_signal_channel_dtype)
+        self._global_channel_to_local_channel = np.array(
+                                    self._global_channel_to_local_channel, dtype='int64')
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        unit_channels = []
+        unit_channels = np.array(unit_channels, dtype=_unit_channel_dtype)
+
+        # deal with nb_segment and t_start/t_stop per segment
+        self._t_starts = {seg_index: 0. for seg_index in range(nb_segment)}
+        self._t_stops = {seg_index: 0. for seg_index in range(nb_segment)}
+        for seg_index in range(nb_segment):
+            for stream_name in stream_names:
+                info = self.signals_info_dict[seg_index, stream_name]
+                t_stop = info['sample_length'] / info['sampling_rate']
+                self._t_stops[seg_index] = max(self._t_stops[seg_index], t_stop)
+
+        # fille into header dict
+        self.header = {}
+        self.header['nb_block'] = 1
+        self.header['nb_segment'] = [nb_segment]
+        self.header['signal_channels'] = sig_channels
+        self.header['unit_channels'] = unit_channels
+        self.header['event_channels'] = event_channels
+
+        # insert some annotation at some place
+        self._generate_minimal_annotations()
+        self._generate_minimal_annotations()
+        block_ann = self.raw_annotations['blocks'][0]
+        block_ann['file_origin'] = self.dirname
+
+        for seg_index in range(nb_segment):
+            seg_ann = block_ann['segments'][seg_index]
+            seg_ann['file_origin'] = self.dirname
+            seg_ann['name'] = "Segment {}".format(seg_index)
+
+            for c in range(sig_channels.size):
+                anasig_an = seg_ann['signals'][c]
+                anasig_an.update(signal_annotations[c])
+
+    def _segment_t_start(self, block_index, seg_index):
+        return 0.
+
+    def _segment_t_stop(self, block_index, seg_index):
+        return self._t_stops[seg_index]
+
+    def _get_signal_size(self, block_index, seg_index, channel_indexes=None):
+        assert channel_indexes is not None
+        stream_name = self._global_channel_to_stream[channel_indexes[0]]
+        memmap = self._memmaps[seg_index, stream_name]
+        return int(memmap.shape[0])
+
+    def _get_signal_t_start(self, block_index, seg_index, channel_indexes):
+        return 0.
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, channel_indexes):
+        assert channel_indexes is not None
+        stream_name = self._global_channel_to_stream[channel_indexes[0]]
+        memmap = self._memmaps[seg_index, stream_name]
+
+        local_chans = self._global_channel_to_local_channel[channel_indexes]
+        if np.all(np.diff(local_chans) == 1):
```

Comment:
If not sroterd then it is a np.take(..) (copy) if sroted then it is a slice (view)
It works in bocth case but one is faster.

Reply:
This is a tick to be faster when a slice is possible. it is quite important for perf. We need to keep this.

Reply:
Yes, that's exactly why I propose to use this performance trick in all cases where it might be applicable, so even when this would require reordering of `local_chans`

Reply:
Postponed for next round of improvements, see #923